state/src/impls/shard_level.rs

```diff
@@ -159,7 +159,12 @@ impl<B: Backend + ShardBackend> ShardLevelState<B> {
                 inputs,
                 outputs,
                 ..
-            } => self.transfer_asset(&transaction, sender, burns, inputs, outputs),
+            } => {
+                if outputs.len() > 126 {
```

Comment:
I think it should be an assertion and verify method should fail.

Reply:
I moved it to verify method